core/payment-driver/zksync/src/zksync/signer.rs

```diff
@@ -105,3 +113,56 @@ fn sign_tx(
     });
     Box::pin(fut)
 }
+
+pub fn encode_signed_tx(raw_tx: &RawTransaction, signature: Vec<u8>, chain_id: u64) -> Vec<u8> {
```

Comment:
does this need to be `pub`?

we can also move this to the base crate to share logic with gnt-driver ( i know i recommended against before, but now i see the need )